tests/v2/08_full_workflow/test_full_workflow_local_processes_v2.py

```diff
@@ -0,0 +1,310 @@
+            json={},
+        )
+        job_data = res.json()
+        debug(job_data)
+        assert res.status_code == 202
+
+        # Check that the workflow execution is complete
+        res = await client.get(
+            f"{PREFIX}/project/{project_id}/job/{job_data['id']}/"
+        )
+        assert res.status_code == 200
+        job_status_data = res.json()
+        debug(job_status_data)
+        assert job_status_data["status"] == "done"
+        debug(job_status_data["end_timestamp"])
+        assert job_status_data["end_timestamp"]
+
+        # Check that the expected files are present
+        working_dir = job_status_data["working_dir"]
+        glob_list = [Path(x).name for x in glob(f"{working_dir}/*")] + [
+            Path(x).name for x in glob(f"{working_dir}/*/*")
+        ]
+
+        must_exist = [
+            "0.log",
+            "0.args.json",
+            IMAGES_FILENAME,
+            HISTORY_FILENAME,
+            FILTERS_FILENAME,
+            WORKFLOW_LOG_FILENAME,
+        ]
+
+        for f in must_exist:
+            assert f in glob_list
+
+        # Check that stderr and stdout are as expected
+        with open(f"{working_dir}/0_non-python/0.log", "r") as f:
+            log = f.read()
+        assert "This goes to standard output" in log
+        assert "This goes to standard error" in log
+
+
+def _sleep_and_return(sleep_time):
+    time.sleep(sleep_time)
+    return 42
+
+
+def test_indirect_shutdown_during_submit(tmp_path):
+
+    shutdown_file = tmp_path / "shutdown"
+    executor = FractalProcessPoolExecutor(shutdown_file=str(shutdown_file))
+
+    res = executor.submit(_sleep_and_return, 100)
+
+    with shutdown_file.open("w"):
+        pass
+    assert shutdown_file.exists()
+
+    time.sleep(2)
+
+    isinstance(res.exception(), BrokenProcessPool)
+    try:
+        res.result()
+    except BrokenProcessPool:
+        pass
+
+
+def wait_one_sec(*args, **kwargs):
+    time.sleep(1)
+    return 42
+
+
+def test_indirect_shutdown_during_map(
+    tmp_path,
+):
+    shutdown_file = tmp_path / "shutdown"
+
+    # NOTE: the executor.map call below is blocking. For this reason, we write
+    # the shutdown file from a subprocess.Popen, so that we can make it happen
+    # during the execution.
+    shutdown_sleep_time = 5
+    tmp_script = (tmp_path / "script.sh").as_posix()
+    debug(tmp_script)
+    with open(tmp_script, "w") as f:
+        f.write(f"sleep {shutdown_sleep_time}\n")
+        f.write(f"cat NOTHING > {shutdown_file.as_posix()}\n")
+
+    tmp_stdout = open((tmp_path / "stdout").as_posix(), "w")
+    tmp_stderr = open((tmp_path / "stderr").as_posix(), "w")
+    subprocess.Popen(
+        shlex.split(f"bash {tmp_script}"),
+        stdout=tmp_stdout,
+        stderr=tmp_stderr,
+    )
+
+    with FractalProcessPoolExecutor(
+        shutdown_file=str(shutdown_file)
+    ) as executor:
+        res = executor.map(wait_one_sec, range(100))
+
+    assert len(list(res)) < 100
```

Comment:
How can we iterate over the result? I would rather expect an exception (either a BrokenProcessPool or JobExecutionError)